### PR TITLE
Automated packaging of proprietary graphic

### PR DESCRIPTION
**Pull request by iartemenko**
Add recipe to make it possible to get graphic package
without building whole rootfs image.

Signed-off-by: Iurii Artemenko <iurii_artemenko@epam.com>

### CODE REVIEW

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-module-egl-headers_1.9.bbappend [hunk 1]
```diff
@@ -7,3 +7,9 @@ SRC_URI_append = " \
     file://0001-EGL-eglext.h-Include-eglmesaext.h.patch \
 "
 
+do_install_append() {
+    if [ ! -d ${DEPLOY_DIR_IMAGE}/xt-rcar ]; then
```
Comment (lorc) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-module-egl-headers_1.9.bbappend]: This check is not needed. `install -d` should not overwrite directory.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-user-module_1.9.bbappend [hunk 2]
```diff
@@ -5,3 +5,9 @@ SRCREV = "${AUTOREV}"
 
 EXTRA_OEMAKE += "PVRSRV_VZ_NUM_OSID=${XT_PVR_NUM_OSID}"
 
+do_install_append() {
+    if [ ! -d ${DEPLOY_DIR_IMAGE}/xt-rcar ]; then
```
Comment (lorc) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-user-module_1.9.bbappend]: The same

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend [hunk 3]
```diff
@@ -40,5 +40,15 @@ populate_vmlinux () {
     find ${STAGING_KERNEL_BUILDDIR} -iname "vmlinux*" -exec mv {} ${DEPLOY_DIR_IMAGE} \; || true
 }
 
-IMAGE_POSTPROCESS_COMMAND += "populate_vmlinux; "
+prepare_graphic () {
```
Comment (lorc) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/images/core-image-weston.bbappend]: I would like to see this in a separate recipe, which has dependencies on other three ones. 
Having separate recipe will allow you to package DDK without rebuilding whole image.

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/rcar-proprietary-graphic/prepare-graphic-package.bb [hunk 4]
```diff
@@ -0,0 +1,18 @@
+LICENSE = "CLOSED"
+
+do_populate_lic[noexec] = "1"
+do_compile[noexec] = "1"
+do_install[depends] += "kernel-module-gles:do_install"
+do_install[depends] += "gles-module-egl-headers:do_install"
+do_install[depends] += "gles-user-module:do_install"
+
+do_install () {
+    if [ -d ${DEPLOY_DIR_IMAGE}/xt-rcar ]; then
```
Comment (lorc) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/rcar-proprietary-graphic/prepare-graphic-package.bb]: Hm, looks you already have such recipe...

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.9.bbappend [hunk 5]
```diff
@@ -0,0 +1,7 @@
+do_install_append() {
+    if [ ! -d ${DEPLOY_DIR_IMAGE}/xt-rcar ]; then
```
Comment (lorc) [inline recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.9.bbappend]: The same about `install -d`

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-module-egl-headers_1.9.bbappend (hunk 1 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-module-egl-headers_1.9.bbappend]: Done

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-user-module_1.9.bbappend (hunk 2 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/gles-module/gles-user-module_1.9.bbappend]: Done

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.9.bbappend (hunk 5 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-kernel/kernel-module-gles/kernel-module-gles_1.9.bbappend]: Done

File: recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/rcar-proprietary-graphic/prepare-graphic-package.bb (hunk 4 above)
Comment (iartemenko) [reply recipes-domd/domd-image-weston/files/meta-xt-prod-extra/recipes-graphics/rcar-proprietary-graphic/prepare-graphic-package.bb]: Done

**Pull request merged**